client/src/components/App.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { ajax } from 'jquery';
+import MarketOrder from './MarketOrder.jsx';
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      orderType: 'Market Order',
+      stock: {},
+      watch: false
+    }
+    this.getRandomStock = this.getRandomStock.bind(this);
+    this.toggleWatch = this.toggleWatch.bind(this);
+    this.renderView = this.renderView.bind(this);
+  }
+
+  componentDidMount() {
+    this.getRandomStock();
+  }
+
+  toggleWatch(e) {
+    e.preventDefault();
+    this.setState((p) => {
+      return {
+        watch: !p.watch
+      }
+    })
+  }
+
+  renderView() {
+    const { orderType } = this.state;
+
+    if (orderType === 'Market Order') {
+      return (
+        <div className="marketOrder">
+          <MarketOrder stock={this.state.stock}/>
+        </div>
+      );
+    }
+  }
+
+  getRandomStock() {
+    ajax({
+      url: '/stocks',
+      method: "GET",
+      success: (data) => {
+        let randomIndex = Math.floor(Math.random() * data.length);
```

Comment:
I had to do a double-take if this could yield an invalid index in the case that Math.random() yielded 1 but looked up that Math.random() yields a number from 0 to 1 not including 1. Nice!

Reply:
Thanks, this was to ensure that the user could not input a decimal value in the form.